### PR TITLE
feat: modifies `useHeadlessFlatTree` return signature to include `getItem` method

### PR DESCRIPTION
**Pull request by bsunderhus**
## Previous Behavior

Currently the only ways to get a headless tree item from the headless tree are:

1. by generating all the items with `.items()` method
2. by simulating a navigation and then using `.getNextNavigableItem()` method
 
## New Behavior

1. modifies the return signature of `useHeadlessFlatTree` to include  `.getItem` method that allow trying to locate the headless tree item

## Related Issue(s)

- Fixes https://github.com/microsoft/fluentui/issues/30985

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">35</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Avatar-mount_base.html">619</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Avatar-mount.html">638</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Button-mount_base.html">304</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Field-mount_base.html">1154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Field-mount.html">1140</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProvider-mount_base.html">718</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProvider-mount.html">702</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-mount_base.html">91</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-mount.html">93</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">35</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">38</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/MakeStyles-mount_base.html">851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/MakeStyles-mount.html">886</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Persona-mount_base.html">1778</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/Persona-mount.html">1725</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/SpinButton-mount_base.html">1376</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/SpinButton-mount.html">1388</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/SwatchPicker-mount_base.html">1522</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31343/perf-test-react-components/SwatchPicker-mount.html">1526</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.139 MB`<br />`275.306 kB` | `1.139 MB`<br />`275.321 kB` | `47 B` <br />`15 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.55 kB`<br />`20.584 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `221.825 kB`<br />`62.568 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.037 kB`<br />`14.418 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1ccf6c23ddc4d125940ce23ea3788d0069665f67'>1ccf6c23ddc4d125940ce23ea3788d0069665f67</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tree/src/index.ts [hunk 1]
```diff
@@ -97,3 +97,4 @@ export type {
 
 export { flattenTree_unstable } from './utils/flattenTree';
 export type { FlattenTreeItem } from './utils/flattenTree';
+export { createHeadlessTree } from './utils/createHeadlessTree';
```
Comment (layershifter) [inline packages/react-components/react-tree/src/index.ts:100]: remove?

**Review by layershifter — APPROVED** [review]

File: change/@fluentui-react-tree-0aa4730f-3deb-451a-b469-e6a748f310d3.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-tree-0aa4730f-3deb-451a-b469-e6a748f310d3.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31343&commitId=98847396aecc1d17694ae805530aca37a70829be&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|5| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**